### PR TITLE
system-variables: add a variable tidb_rc_write_check_ts

### PR DESCRIPTION
**Pull request by TonsnakeLin**
Signed-off-by: TonsnakeLin <lpbgytong@163.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TonsnakeLin** [comment]
@cfzjywxk @ekexium

File: system-variables.md [hunk 1]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
 - 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下读写冲突较少的场景，开启此变量可以避免获取全局 timestamp 带来的延迟和开销，并优化事务内读语句延迟。
 - 如果读写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
 
+### `tidb_rc_write_check_ts` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+> **警告：**
+>
+> - 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容，开启 `tidb_rc_write_check_ts` 的请求无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，请勿同时开启两项特性。
```
Comment (ekexium) [inline system-variables.md:1934]: 同时开启两个会发生啥要说一下吗

**Review by ekexium — COMMENTED** [review]
LGTM. Should we add corresponding content in the `Read-committed` section, along with those with `rc_read_check_ts`?

File: system-variables.md [hunk 2]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下点写冲突较少的场景，开启此变量可以避免点写语句获取全局 timestamp 带来的延迟和开销。目前点写语句的类型控制在 `UPDATE`、`DELETE`、`SELECT ...... FOR UPDATE` 三种类型。点写语句是指将主键或者唯一键作为过滤条件且最终执行算子包含 `POINT-GET` 的写语句。
```
Comment (TomShawn) [inline system-variables.md:1939]:
```suggestion
- 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下点写冲突较少的场景。开启此变量可以避免点写语句获取全局时间戳带来的延迟和开销。目前该变量适用的点写语句包括 `UPDATE`、`DELETE`、`SELECT ...... FOR UPDATE` 三种类型。点写语句是指将主键或者唯一键作为过滤条件且最终执行算子包含 `POINT-GET` 的写语句。
```

File: system-variables.md [hunk 3]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
...
+- 如果点写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
```
Comment (TomShawn) [inline system-variables.md:1940]:
```suggestion
- 如果点写冲突较为严重，开启此变量会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
```

File: system-variables.md (hunk 1 above)
Comment (TomShawn) [inline system-variables.md:1934]:
```suggestion
> - 本变量对应的特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，请求无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启两项特性。
```

**Comment by TomShawn** [comment]
> LGTM. Should we add corresponding content in the `Read-committed` section, along with those with `rc_read_check_ts`?

Agree

File: transaction-isolation-levels.md [hunk 4]
```diff
@@ -69,6 +69,15 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
 
 在使用 `READ-COMMITTED` 隔离级别且单个事务中 `SELECT` 语句较多、读写冲突较少的场景，可通过开启此变量来避免获取全局 timestamp 带来的延迟和开销。
 
+从 v6.3.0 版本开始，TiDB 支持使用系统变量 [`tidb_rc_write_check_ts`](/system-variables.md#tidb_rc_write_check_ts-从-v630-版本开始引入) 对点写冲突较少情况下优化时间戳的获取。开启此变量后，点写语句会尝试使用当前事务有效的时间戳进行数据读取和加锁操作，且在读取数据时按照开启 [`tidb_rc_read_check_ts`](/system-variables.md#tidb_rc_read_check_ts-从-v600-版本开始引入) 方式读取数据。目前点写语句的类型控制在 `UPDATE`、`DELETE`、`SELECT ...... FOR UPDATE` 三种类型。点写语句是指将主键或者唯一键作为过滤条件且最终执行算子包含 `POINT-GET` 的写语句。上述三种点写语句的共同点是会先根据 key 值做点查，如果 key 存在再加锁，如果不存在则直接返回空集。
```
Comment (TomShawn) [inline transaction-isolation-levels.md:72]:
```suggestion
从 v6.3.0 版本开始，TiDB 支持通过开启系统变量 [`tidb_rc_write_check_ts`](/system-variables.md#tidb_rc_write_check_ts-从-v630-版本开始引入) 对点写冲突较少情况下优化时间戳的获取。开启此变量后，点写语句会尝试使用当前事务有效的时间戳进行数据读取和加锁操作，且在读取数据时按照开启 [`tidb_rc_read_check_ts`](/system-variables.md#tidb_rc_read_check_ts-从-v600-版本开始引入) 的方式读取数据。目前该变量适用的点写语句包括 `UPDATE`、`DELETE`、`SELECT ...... FOR UPDATE` 三种类型。点写语句是指将主键或者唯一键作为过滤条件且最终执行算子包含 `POINT-GET` 的写语句。目前这三种点写语句的共同点是会先根据 key 值做点查，如果 key 存在再加锁，如果不存在则直接返回空集。
```

File: system-variables.md [hunk 5]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
 - 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下读写冲突较少的场景，开启此变量可以避免获取全局 timestamp 带来的延迟和开销，并优化事务内读语句延迟。
 - 如果读写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
 
+### `tidb_rc_write_check_ts` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+> **警告：**
+>
+> - 本变量对应的特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，请求无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启两项特性。
```
Comment (TomShawn) [inline system-variables.md:1934]: 是什么请求呢？无法使用 replica read 进行副本读取操作吗？

File: transaction-isolation-levels.md [hunk 6]
```diff
@@ -69,6 +69,15 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
...
+
+- 如果点写语句整个读取过程没有遇到更新的数据版本，则继续使用当前事务的时间戳进行加锁。
+    - 如果加锁过程中遇到因时间戳旧而导致的写冲突，则重新获取最新的全局时间戳重新进行加锁。
+    - 如果加锁过程中没有遇到写冲突和其他错误，则加锁成功。
+- 如果读取过程中遇到更新的数据版本，则尝试重新获取一个新的时间戳重试此语句。
+ 
+在使用 `READ-COMMITTED` 隔离级别且单个事务中点写语句较多、点写冲突较少的场景，可通过开启此变量来避免获取全局 timestamp 带来的延迟和开销。
```
Comment (TomShawn) [inline transaction-isolation-levels.md:79]:
```suggestion
- 如果点写语句的整个读取过程中没有遇到更新的数据版本，则继续使用当前事务的时间戳进行加锁。
    - 如果加锁过程中遇到因时间戳旧而导致的写冲突，则重新获取最新的全局时间戳重新进行加锁。
    - 如果加锁过程中没有遇到写冲突和其他错误，则加锁成功。
- 如果读取过程中遇到更新的数据版本，则尝试重新获取一个新的时间戳重试此语句。
 
在使用 `READ-COMMITTED` 隔离级别且单个事务中点写语句较多、点写冲突较少的场景，可通过开启此变量来避免获取全局时间戳带来的延迟和开销。
```

File: system-variables.md (hunk 5 above)
Comment (TonsnakeLin) [reply system-variables.md:1934]: 客户端发送的任意请求都无法使用 replica read。

Comment (shichun-0415) [inline system-variables.md:1934]:
```suggestion
> - 本变量对应的特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `replica-read` 和 `replica-read`。
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 7]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
 - 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下读写冲突较少的场景，开启此变量可以避免获取全局 timestamp 带来的延迟和开销，并优化事务内读语句延迟。
 - 如果读写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
 
+### `tidb_rc_write_check_ts` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+> **警告：**
+>
+> - 本变量对应的特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `replica-read` 和 `replica-read`。
```
Comment (TomShawn) [inline system-variables.md:1934]:
```suggestion
> - 本变量对应的特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `replica-read` 和 `tidb_rc_write_check_ts`。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Review by ekexium — APPROVED** [review]

File: system-variables.md [hunk 8]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
 - 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下读写冲突较少的场景，开启此变量可以避免获取全局 timestamp 带来的延迟和开销，并优化事务内读语句延迟。
 - 如果读写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
 
+### `tidb_rc_write_check_ts` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+> **警告：**
+>
+> - 本变量对应的特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `replica-read` 和 `tidb_rc_write_check_ts`。
```
Comment (TomShawn) [inline system-variables.md:1934]:
```suggestion
> - 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `tidb_rc_write_check_ts` 和 `replica-read`。
```

File: system-variables.md [hunk 9]
```diff
@@ -1927,6 +1927,18 @@ explain select * from t where age=5;
 - 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下读写冲突较少的场景，开启此变量可以避免获取全局 timestamp 带来的延迟和开销，并优化事务内读语句延迟。
 - 如果读写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
 
+### `tidb_rc_write_check_ts` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+> **警告：**
+>
+> - 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `tidb_rc_write_check_ts` 和 `replica-read`。
```
Comment (TomShawn) [inline system-variables.md:1934]:
```suggestion
> 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `tidb_rc_write_check_ts` 和 `replica-read`。
```

File: system-variables.md [hunk 10]
```diff
@@ -1920,13 +1920,26 @@ explain select * from t where age=5;
 >
 > - 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容，开启 `tidb_rc_read_check_ts` 的读请求无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，请勿同时开启两项特性。
 > - 如果客户端使用游标操作，建议不开启 `tidb_rc_read_check_ts` 这一特性，避免前一批返回数据已经被客户端使用而语句最终会报错的情况。
+> - tidb_rc_read_check_ts 在 v6.3.0 版本中由 GLOBAL/SESSION 修改为 INSTANCE 级别。
 
-- 作用域：SESSION | GLOBAL
-- 是否持久化到集群：是
+- 作用域：GLOBAL
```
Comment (shichun-0415) [inline system-variables.md:1925]:
```suggestion
- 作用域：INSTANCE
```

File: system-variables.md [hunk 11]
```diff
@@ -1920,13 +1920,26 @@ explain select * from t where age=5;
...
+- 是否持久化到集群：否，仅作用于当前连接的 TiDB 实例
 - 默认值：`OFF`
 - 该变量用于优化时间戳的获取，适用于悲观事务 `READ-COMMITTED` 隔离级别下读写冲突较少的场景，开启此变量可以避免获取全局 timestamp 带来的延迟和开销，并优化事务内读语句延迟。
 - 如果读写冲突较为严重，开启此功能会增加额外开销和延迟，造成性能回退。更详细的说明，请参考[读已提交隔离级别 (Read Committed) 文档](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
 
+### `tidb_rc_write_check_ts` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+> **警告：**
+>
+> 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入)，因此请勿同时开启 `tidb_rc_write_check_ts` 和 `replica-read`。
```
Comment (shichun-0415) [inline system-variables.md:1935]:
```suggestion
> 该特性与 [`replica-read`](#tidb_replica_read-从-v40-版本开始引入) 尚不兼容。开启本变量后，客户端发送的所有请求都将无法使用 `replica-read`，因此请勿同时开启 `tidb_rc_write_check_ts` 和 `replica-read`。
```

File: transaction-isolation-levels.md [hunk 12]
```diff
@@ -69,6 +69,15 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
 
 在使用 `READ-COMMITTED` 隔离级别且单个事务中 `SELECT` 语句较多、读写冲突较少的场景，可通过开启此变量来避免获取全局 timestamp 带来的延迟和开销。
 
+从 v6.3.0 版本开始，TiDB 支持通过开启系统变量 [`tidb_rc_write_check_ts`](/system-variables.md#tidb_rc_write_check_ts-从-v630-版本开始引入) 对点写冲突较少情况下优化时间戳的获取。开启此变量后，点写语句会尝试使用当前事务有效的时间戳进行数据读取和加锁操作，且在读取数据时按照开启 [`tidb_rc_read_check_ts`](/system-variables.md#tidb_rc_read_check_ts-从-v600-版本开始引入) 的方式读取数据。目前该变量适用的点写语句包括 `UPDATE`、`DELETE`、`SELECT ...... FOR UPDATE` 三种类型。点写语句是指将主键或者唯一键作为过滤条件且最终执行算子包含 `POINT-GET` 的写语句。目前这三种点写语句的共同点是会先根据 key 值做点查，如果 key 存在再加锁，如果不存在则直接返回空集。
+
+- 如果点写语句的整个读取过程中没有遇到更新的数据版本，则继续使用当前事务的时间戳进行加锁。
+    - 如果加锁过程中遇到因时间戳旧而导致的写冲突，则重新获取最新的全局时间戳重新进行加锁。
```
Comment (shichun-0415) [inline transaction-isolation-levels.md:75]:
```suggestion
    - 如果加锁过程中遇到因时间戳旧而导致写冲突，则重新获取最新的全局时间戳进行加锁。
```

File: transaction-isolation-levels.md [hunk 13]
```diff
@@ -69,6 +69,15 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
...
+    - 如果加锁过程中没有遇到写冲突和其他错误，则加锁成功。
```
Comment (shichun-0415) [inline transaction-isolation-levels.md:76]:
```suggestion
    - 如果加锁过程中没有遇到写冲突或其他错误，则加锁成功。
```

**Pull request merged**